ckks/polynomial_evaluation.go

```diff
@@ -188,13 +229,47 @@ func (eval *evaluator) evaluatePolyVector(ct0 *Ciphertext, pol polynomialVector,
 	return opOut, err
 }
 
-func computePowerBasis(n int, C map[int]*Ciphertext, scale float64, basis PolynomialBasis, evaluator *evaluator) (err error) {
+// PowerBasis is a struct storing powers of a ciphertext.
+type PowerBasis struct {
+	Value map[int]*Ciphertext
+	Basis PolynomialBasis
+}
+
+// NewPowerBasis creates a new PowerBasis.
+func NewPowerBasis(ct *Ciphertext, Basis PolynomialBasis) (p *PowerBasis) {
```

Comment:
It might be unclear that this outputs a non-initialized basis.

Reply:
- If no power-basis is given to poly eval -> evaluator looks what ring is used to decide lazy or not
- Change PolynomialBasis to BasisType
- Add flag lazy to NewPowerBasis